### PR TITLE
added ruby implementation of Armstrong number

### LINKED ISSUE #1564: Creation of Armstrong number implementation in ruby
There is no implementation of Armstrong number in ruby. So  I want to create a ruby implementation of this problem.

### PR DESCRIPTION
**Pull request by darkdebo**
Fixes #1564

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

all checkbox are checked for my pull request (PR) which is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added the Armstrong_Number.rb file which contains the ruby implementation of the Armstrong number.
- Changed the Armstrong number readme file with ruby implementation link and run online information added

#### Languages Used:
- Ruby
- 

#### Files Added:
- Armstrong_Number.rb

#### Files Changed
- changed the readme.md file

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks for your help!

### CODE REVIEW

File: Armstrong_Number/Armstrong_Number.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+#To calculate the order of the number
+def order(num)
+	# variable to store of the number 
+	count = 0
+	while (num != 0) 
+		count = count + 1
+    num = num / 10
+  end
+  return count
```
Comment (asha15) [inline Armstrong_Number/Armstrong_Number.rb]: correct the indentation

File: Armstrong_Number/Armstrong_Number.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# Function to check whether the given number is 
+# Armstrong number or not 
+def Armstrong (num) 
+	count = order(num) 
+	temp = num 
+	sum1 = 0
+	while (temp != 0)
+		d = temp % 10
+		sum1 = sum1 + d**count
+    temp = temp / 10
```
Comment (asha15) [inline Armstrong_Number/Armstrong_Number.rb]: correct indentation

File: Armstrong_Number/Armstrong_Number.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+	# If condition satisfies 
+  boolean=false
+
+  if(num!=sum1)
+    return boolean
+  else if(num==sum1)
+    boolean=true
+    return boolean
```
Comment (asha15) [inline Armstrong_Number/Armstrong_Number.rb]: here too
please keep space before and after "=" and "=="

File: Armstrong_Number/Armstrong_Number.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  if(num!=sum1)
+    return boolean
+  else if(num==sum1)
+    boolean=true
+    return boolean 
+  end 
+  end
+end
+
```
Comment (asha15) [inline Armstrong_Number/Armstrong_Number.rb]: remove this extra line

File: Armstrong_Number/Armstrong_Number.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
```
Comment (asha15) [inline Armstrong_Number/Armstrong_Number.rb]: remove the extra lines

File: Armstrong_Number/Armstrong_Number.rb (hunk 1 above)
Comment (darkdebo) [reply Armstrong_Number/Armstrong_Number.rb]: resolved

File: Armstrong_Number/Armstrong_Number.rb (hunk 5 above)
Comment (darkdebo) [reply Armstrong_Number/Armstrong_Number.rb]: removed

**Comment by darkdebo** [comment]
> comment:
> 
> * corrected the indentation
> * provided two sample input and output
> * remove the readme.md file changes.

File: Armstrong_Number/Armstrong_Number.rb [hunk 6]
```diff
@@ -0,0 +1,53 @@
+#A positive integer of n digits is called an Armstrong number of order n (order is number of digits) if
```
Comment (MastersAbh) [inline Armstrong_Number/Armstrong_Number.rb]: Please follow 4 space indentation across the file

**Comment by darkdebo** [comment]
Please review it @MastersAbh

File: Armstrong_Number/Armstrong_Number.rb [hunk 7]
```diff
@@ -0,0 +1,48 @@
+    #A positive integer of n digits is called an Armstrong number of order n (order is number of digits) if
+    #abcd... = pow(a,n) + pow(b,n) + pow(c,n) + pow(d,n) + .... 
+    
+    #To calculate the order of the number
+    def order(num)
+        # variable to store of the number
```
Comment (MastersAbh) [inline Armstrong_Number/Armstrong_Number.rb]: This function still has 2 space indentation. Please make it 4

File: Armstrong_Number/Armstrong_Number.rb [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    # Armstrong number or not 
+    def Armstrong (num) 
+        count = order(num) 
+        temp = num 
+        sum1 = 0
+        while (temp != 0)
+                  d = temp % 10
+                  sum1 = sum1 + d**count
+        temp = temp / 10
+      end
```
Comment (MastersAbh) [inline Armstrong_Number/Armstrong_Number.rb]: indentation of "end"

File: Armstrong_Number/Armstrong_Number.rb [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    end
+    
+    # Function to check whether the given number is 
+    # Armstrong number or not 
+    def Armstrong (num) 
+        count = order(num) 
+        temp = num 
+        sum1 = 0
+        while (temp != 0)
+                  d = temp % 10
```
Comment (MastersAbh) [inline Armstrong_Number/Armstrong_Number.rb]: looks like 8 space indentation inside while loop

File: Armstrong_Number/Armstrong_Number.rb [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+      end
+    
+        # If condition satisfies 
+      boolean=false
+    
+      if (num != sum1)
+            return boolean
+      else if(num == sum1)
+                boolean=true
+                return boolean
```
Comment (MastersAbh) [inline Armstrong_Number/Armstrong_Number.rb]: please fix indentation

**Comment by darkdebo** [comment]
Done with indentation.

File: Armstrong_Number/Armstrong_Number.rb [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+      end
+    end
+    
+    # Driver Program 
+    num = 153
+    print(Armstrong(num))
+    
+    #True
+    
+    num = 134
```
Comment (MastersAbh) [inline Armstrong_Number/Armstrong_Number.rb]: Please take dynamic inputs.
Also, you need to indent only the blocks within the code. Please remove the initial margin given to the code.

File: Armstrong_Number/Armstrong_Number.rb [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+                d = temp % 10
+                sum1 = sum1 + d**count
+        temp = temp / 10
+        end
+    
+        # If condition satisfies 
+      boolean=false
+    
+      if (num != sum1)
+            return boolean
```
Comment (MastersAbh) [inline Armstrong_Number/Armstrong_Number.rb]: There is 6 space instead of 4

File: Armstrong_Number/Armstrong_Number.rb [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+    
+    # Function to check whether the given number is 
+    # Armstrong number or not 
+    def Armstrong (num) 
+        count = order(num) 
+        temp = num 
+        sum1 = 0
+        while (temp != 0)
+                d = temp % 10
+                sum1 = sum1 + d**count
```
Comment (MastersAbh) [inline Armstrong_Number/Armstrong_Number.rb]: Fix indentation

File: Armstrong_Number/Armstrong_Number.rb [hunk 14]
```diff
@@ -0,0 +1,48 @@
+    #A positive integer of n digits is called an Armstrong number of order n (order is number of digits) if
+    #abcd... = pow(a,n) + pow(b,n) + pow(c,n) + pow(d,n) + .... 
+    
+    #To calculate the order of the number
+      def order(num)
+          # variable to store of the number 
+          count = 0
+          while (num != 0) 
+                  count = count + 1
```
Comment (MastersAbh) [inline Armstrong_Number/Armstrong_Number.rb]: Fix indentation

**Comment by darkdebo** [comment]
Dynamic input taken.Please check it @MastersAbh

File: Armstrong_Number/Armstrong_Number.rb [hunk 15]
```diff
@@ -0,0 +1,54 @@
+#A positive integer of n digits is called an Armstrong number of order n (order is number of digits) if
```
Comment (jainaman224) [inline Armstrong_Number/Armstrong_Number.rb]: Ideally you should have space after `#`.

File: Armstrong_Number/Armstrong_Number.rb [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+#abcd... = pow(a,n) + pow(b,n) + pow(c,n) + pow(d,n) + .... 
+
+#To calculate the order of the number
+def order(num)
+	# variable to store of the number
```
Comment (jainaman224) [inline Armstrong_Number/Armstrong_Number.rb]: 4 space indentation is not followed.

File: Armstrong_Number/Armstrong_Number.rb [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+		temp = temp / 10
+	end
+
+	# If condition satisfies 
+	
+
+	if (num != sum1)
+		return boolean
+	else if(num == sum1)
+		boolean=true
```
Comment (jainaman224) [inline Armstrong_Number/Armstrong_Number.rb]: Add space around operators like `=`, `+`.

File: Armstrong_Number/Armstrong_Number.rb (hunk 16 above)
Comment (darkdebo) [reply Armstrong_Number/Armstrong_Number.rb]: Have I need indent the whole code with comments 4 indent or only this part.

**Comment by darkdebo** [comment]
Everything is modified, please check it

File: Armstrong_Number/Armstrong_Number.rb [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+	    count = count + 1
+		num = num / 10
+	end
+	return count 
+end
+
+# Function to check whether the given number is 
+# Armstrong number or not 
+def Armstrong (num) 
+	count = order(num)
```
Comment (jainaman224) [inline Armstrong_Number/Armstrong_Number.rb]: Use 4 space indentation.

File: Armstrong_Number/Armstrong_Number.rb [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+
+#num = 134
+#print(Armstrong(num))
+
+#False
+
+#num = 370
+#print(Armstrong(num))
+
+#True
```
Comment (jainaman224) [inline Armstrong_Number/Armstrong_Number.rb]: Add a single new line at the end of file.

File: Armstrong_Number/Armstrong_Number.rb (hunk 18 above)
Comment (jainaman224) [reply Armstrong_Number/Armstrong_Number.rb]: Make this changes throughout the file.

Comment (darkdebo) [reply Armstrong_Number/Armstrong_Number.rb]: Means whole file need to be 4 space indent including function.

**Comment by darkdebo** [comment]
please review it @jainaman224  if any indentation problem.

File: Armstrong_Number/Armstrong_Number.rb [hunk 20]
```diff
@@ -0,0 +1,54 @@
+# A positive integer of n digits is called an Armstrong number of order n (order is number of digits) if
+# abcd... = pow(a,n) + pow(b,n) + pow(c,n) + pow(d,n) + .... 
+
+# To calculate the order of the number
+def order(num)
+	# variable to store of the number 
+    count = 0
+	while (num != 0) 
+	    	count = count + 1
+		num = num / 10
```
Comment (MastersAbh) [inline Armstrong_Number/Armstrong_Number.rb]: This is still not 4 space indent

File: Armstrong_Number/Armstrong_Number.rb [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+# Function to check whether the given number is 
+# Armstrong number or not 
+def Armstrong (num) 
+	count = order(num) 
+	temp = num 
+	sum1 = 0
+	boolean = false
+	while (temp != 0)
+		d = temp % 10
+		sum1 = sum1 + d**count
```
Comment (MastersAbh) [inline Armstrong_Number/Armstrong_Number.rb]: Please use 4 space instead of TAB

**Comment by darkdebo** [comment]
@MastersAbh @jainaman224 please check it.
@shlok007.

File: Armstrong_Number/Armstrong_Number.rb [hunk 22]
```diff
@@ -1,42 +1,41 @@
...
+  boolean = false
+  while (temp != 0)
+    d = temp % 10
+    sum1 = sum1 + d ** count
+    temp = temp / 10
+  end
+
+  # If condition satisfies
+
+  if (num != sum1)
```
Comment (shlok007) [inline Armstrong_Number/Armstrong_Number.rb]: how about replacing the whole block with`return num == sum1` ?

Comment (darkdebo) [reply Armstrong_Number/Armstrong_Number.rb]: may be good ok i give a try

Comment (darkdebo) [reply Armstrong_Number/Armstrong_Number.rb]: @shlok007 done

**Comment by darkdebo** [comment]
Please review it

File: Armstrong_Number/Armstrong_Number.rb [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+  sum1 = 0
+  boolean = false
+  while (temp != 0)
+    d = temp % 10
+    sum1 = sum1 + d ** count
+    temp = temp / 10
+  end
+
+  # If condition satisfies
+
```
Comment (asha15) [inline Armstrong_Number/Armstrong_Number.rb]: remove this line

Comment (darkdebo) [reply Armstrong_Number/Armstrong_Number.rb]: only this line

Comment (darkdebo) [reply Armstrong_Number/Armstrong_Number.rb]: @asha15 please check and tell me if anything more needed to remove. readme 
files changes are removed.

**Comment by darkdebo** [comment]
I close this pull request and open a new one with armstrong number ruby file,please check that one
@asha15 https://github.com/jainaman224/Algo_Ds_Notes/pull/2501

**Pull request closed without merging**